akka-cluster/src/main/scala/akka/cluster/sbr/DowningStrategy.scala

```diff
@@ -321,6 +326,8 @@ import akka.coordination.lease.scaladsl.Lease
       case DownIndirectlyConnected                   => ReverseDownIndirectlyConnected
       case AcquireLeaseAndDownIndirectlyConnected(_) => ReverseDownIndirectlyConnected
       case ReverseDownIndirectlyConnected            => DownIndirectlyConnected
+      case DownSelfQuarantinedByRemote =>
+        throw new IllegalArgumentException("Not expected to ever try to reverse DownSelfQuarantinedByRemote")
```

Comment:
The name of this method is misleading as the goal is not to reverse a decision. The exception message is also misleading in that sense. 

We only call that method when we fail to acquire the lease, so we need to convert the `decision` to something that makes sense for the side that didn't get the lease. 

When we reach that point, the only possible decisions are `AcquireLeaseAndDownUnreachable` and `AcquireLeaseAndDownIndirectlyConnected` because that was already narrow down to `AcquireLeaseDecision`.

This could be refactored to: 

```scala
def reverseDecision(decision: AcquireLeaseDecision): Decision = {
  decision match {
    case AcquireLeaseAndDownUnreachable(_)         => DownReachable
    case AcquireLeaseAndDownIndirectlyConnected(_) => ReverseDownIndirectlyConnected
  }
}
```

Reply:
I think it says exactly what it does, it reverses a decision, `DownIndirectlyConnected` becomes `ReverseDownIndirectlyConnected`. I'd prefer to leave this as is and keep the PR to be about one isolated thing, please open up a separate PR if you really think it needs renaming/cleanup.

Reply:
Ok, we can do the clean-up on another PR.

Reply:
I have the code ready and was planning to send a PR on top of this PR, but this PR is from a fork. So, I will wait for this to be merged.